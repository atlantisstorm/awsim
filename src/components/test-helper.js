import React from 'react';
//import { render } from "@testing-library/react";
import Context from "../store/context";
import { initialiseReducer } from "../store/reducer";

export const RenderWithProvider = (props) => {
  const { children, testState, testDispatch }  = props;
  // Note: we pass any supplied testState to initialiseReducer rather than switching it like
  // we do for dispatch because of scoping.  Basically each rerender after a click/select
  // update event would mean we refresh again with the original data.
  const { state, dispatch: iDispatch } = initialiseReducer({ initialState: testState });
  const dispatch = testDispatch ? testDispatch : iDispatch;
  return (
    <Context.Provider value={{ state, dispatch }}>
      { children }
    </Context.Provider> 
  )
}