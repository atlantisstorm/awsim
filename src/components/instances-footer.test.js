import React from 'react';
import { render } from "@testing-library/react";
import InstancesFooter from './instances-footer';
import { RenderWithProvider } from './test-helper';

describe('InstancesFooter component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <RenderWithProvider>
        <table>
          <InstancesFooter />
        </table>
      </RenderWithProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});