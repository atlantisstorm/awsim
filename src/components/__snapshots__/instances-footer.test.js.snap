// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstancesFooter component should match snapshot 1`] = `
<DocumentFragment>
  <table>
    <tfoot>
      <tr>
        <td
          colspan="12"
        >
          0 instances found.
        </td>
      </tr>
    </tfoot>
  </table>
</DocumentFragment>
`;
